### PR TITLE
Integration with External API

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
Integration with external Api (Open Library) and new route for get information of a book through his ISBN code.
### ScreenShoots
image
### Trello
https://trello.com/c/fGWfmGiM/36-integraci%C3%B3n-api-externa
### Notes
The new endpoint is "/api/v1/query/book/:isbn".

### CODE REVIEW

File: app/services/openlibrary_service.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class OpenLibraryService
+  include HTTParty
+  base_uri 'openlibrary.org'
+  def initialize
```
Comment (facucompe) [inline app/services/openlibrary_service.rb]: There is no need to do this in the initialize

File: app/services/openlibrary_service.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    @options = { query: { format: 'json', jscmd: 'data' } }
+  end
+
+  def book_info(isbn)
+    ibsn_formatted = 'ISBN:'.concat(isbn)
+    @options[:query][:bibkeys] = ibsn_formatted
```
Comment (facucompe) [inline app/services/openlibrary_service.rb]: Create a method that returns the request body

File: config/routes.rb [hunk 3]
```diff
@@ -5,6 +5,7 @@
      resources :users do
       resources :rents, only: [:create, :index]
      end
+     get '/query/book/:isbn', to: 'query_book#query_by_isbn'
```
Comment (facucompe) [inline config/routes.rb:9]: Create a route inside `/books` routes. Something like `books/:id/isbn_info`

File: app/controllers/api/v1/query_book_controller.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require 'openlibrary_service'
+module Api
+  module V1
+    class QueryBookController < ApplicationController
+      def query_by_isbn
+        service = OpenLibraryService.new
```
Comment (facucompe) [inline app/controllers/api/v1/query_book_controller.rb:6]: Remove this line

File: app/controllers/api/v1/query_book_controller.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+        external_response = service.book_info(params[:isbn])
```
Comment (facucompe) [inline app/controllers/api/v1/query_book_controller.rb:7]: `OpenLibraryService.new.info(params[:isbn])`

File: app/controllers/api/v1/query_book_controller.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        render json: external_response
```
Comment (facucompe) [inline app/controllers/api/v1/query_book_controller.rb]: What if response it's a failure?

**Review by facucompe — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/query_book_controller.rb (hunk 6 above)
Comment (mcolombo87) [reply app/controllers/api/v1/query_book_controller.rb]: Done!

File: app/services/openlibrary_service.rb (hunk 1 above)
Comment (mcolombo87) [reply app/services/openlibrary_service.rb]: Done!

File: app/services/openlibrary_service.rb (hunk 2 above)
Comment (mcolombo87) [reply app/services/openlibrary_service.rb]: Done!

File: app/controllers/api/v1/query_book_controller.rb [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+module Api
+  module V1
+    class QueryBookController < ApplicationController
+      def query_by_isbn
+        service = OpenLibraryService.new
+        external_response = service.book_info(params[:isbn])
+        if external_response.status == :ok
+          render json: service.parsing_isbn_response(external_response)
+        else
+          render json: external_response.message, status: :bad_request
```
Comment (floresv) [inline app/controllers/api/v1/query_book_controller.rb:11]: return first errors and at the end OK response. Use one line conditionals.

File: app/controllers/api/v1/query_book_controller.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
+require 'openlibrary_service'
+module Api
+  module V1
+    class QueryBookController < ApplicationController
+      def query_by_isbn
+        service = OpenLibraryService.new
+        external_response = service.book_info(params[:isbn])
+        if external_response.status == :ok
```
Comment (facucompe) [inline app/controllers/api/v1/query_book_controller.rb]: `external_response.ok?`

**Review by facucompe — DISMISSED** [review]
Try to change the route

**Review by facucompe — APPROVED** [review]

**Pull request merged**